modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusRoutes.scala

```diff
@@ -1,24 +1,59 @@
 package org.tessellation.sdk.infrastructure.consensus
 
+import cats.Applicative
 import cats.effect.Async
-import cats.syntax.all._
-import cats.{Order, Show}
+import cats.kernel.Next
+import cats.syntax.contravariantSemigroupal._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
 
-import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.peer.PeerId
+import org.tessellation.sdk.domain.cluster.services.Cluster
 import org.tessellation.sdk.infrastructure.consensus.registration.RegistrationResponse
 
-import io.circe.{Decoder, Encoder}
+import eu.timepit.refined.auto._
+import io.circe.Encoder
 import org.http4s.HttpRoutes
 import org.http4s.circe.CirceEntityCodec._
 import org.http4s.dsl.Http4sDsl
+import org.http4s.server.Router
 
-class ConsensusRoutes[F[_]: Async: KryoSerializer, Key: Order: Show: Encoder: Decoder](
-  storage: ConsensusStorage[F, _, Key, _]
+class ConsensusRoutes[F[_]: Async, Key: Encoder: Next](
+  cluster: Cluster[F],
+  storage: ConsensusStorage[F, _, Key, _],
+  selfId: PeerId
 ) extends Http4sDsl[F] {
 
-  val routes: HttpRoutes[F] = HttpRoutes.of[F] {
-    case GET -> Root / "consensus" / "registration" =>
+  private val prefixPath = "/consensus"
+
+  private val p2p: HttpRoutes[F] = HttpRoutes.of[F] {
+    case GET -> Root / "registration" =>
       storage.getOwnRegistration.map(RegistrationResponse(_)).flatMap(Ok(_))
   }
 
+  private val public: HttpRoutes[F] = HttpRoutes.of[F] {
+    case GET -> Root / "latest/peers" =>
```

Comment:
WDYT about `peers/latest`? Because `latest` is a sub resource of `peers`. Not the other way.